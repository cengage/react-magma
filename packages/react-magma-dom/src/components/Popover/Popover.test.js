import React from 'react';
import { Popover, PopoverContent, PopoverTrigger } from '.';
import { act, render } from '@testing-library/react';

describe('Popover', () => {
  it('should find element by testId', () => {
    const testId = 'test-id';
    const { getByTestId } = render(
      <Popover testId={testId}>
        <PopoverTrigger />
        <PopoverContent>
          <span>Content</span>
        </PopoverContent>
      </Popover>
    );

    expect(getByTestId(testId)).toBeInTheDocument();
  });

  it('should render the popover component, positioned bottom by default', async () => {
    const testId = 'test-id';

    const { container, getByText, getByTestId } = render(
      <Popover testId={testId}>
        <PopoverTrigger />
        <PopoverContent>
          <span>Content</span>
        </PopoverContent>
      </Popover>
    );
    const popoverTrigger = container.querySelector('button');

    expect(container).toBeInTheDocument();
    expect(popoverTrigger).toBeInTheDocument();

    await act(async () => {
      popoverTrigger.click();
    });

    const popoverContent = getByTestId('popoverContent');
    const popoverContentDialog = container.querySelector('div[role="dialog"]');
    const spanContent = getByText('Content');

    expect(popoverContent).toBeInTheDocument();
    expect(popoverContentDialog).toBeInTheDocument();
    expect(spanContent).toBeInTheDocument();

    expect(popoverContent).toHaveAttribute('data-popover-placement', 'bottom');
    expect(popoverContent).toMatchSnapshot();
  });
});
