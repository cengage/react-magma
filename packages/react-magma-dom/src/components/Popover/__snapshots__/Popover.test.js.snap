// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Popover should render the popover with isInverse prop 1`] = `
.emotion-1 {
  background: #292F7C;
  border: 1px solid rgba(255,255,255,0.5);
  border-radius: 8px;
  box-shadow: 0 2px 6px 0 rgba(0,0,0,0.18);
  color: #FFFFFF;
  font-family: "Work Sans",Helvetica,sans-serif;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  overflow: visible;
  padding-left: 0;
  position: relative;
  text-align: left;
  width: auto;
  position: relative;
  background: #292F7C;
  display: none;
  max-height: 100%;
  opacity: 0;
  outline: 0;
  -webkit-transition: opacity 0.3s;
  transition: opacity 0.3s;
  white-space: nowrap;
  border: 1px solid #5D65CB;
  width: 100%;
  max-width: 300px;
}

.emotion-3 {
  width: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.emotion-5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  overflow-y: auto;
  padding: 16px;
  height: inherit;
  white-space: normal;
}

.emotion-1 {
  background: #292F7C;
  border: 1px solid rgba(255,255,255,0.5);
  border-radius: 8px;
  box-shadow: 0 2px 6px 0 rgba(0,0,0,0.18);
  color: #FFFFFF;
  font-family: "Work Sans",Helvetica,sans-serif;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  overflow: visible;
  padding-left: 0;
  position: relative;
  text-align: left;
  width: auto;
  position: relative;
  background: #292F7C;
  display: none;
  max-height: 100%;
  opacity: 0;
  outline: 0;
  -webkit-transition: opacity 0.3s;
  transition: opacity 0.3s;
  white-space: nowrap;
  border: 1px solid #5D65CB;
  width: 100%;
  max-width: 300px;
}

.emotion-3 {
  width: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.emotion-5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  overflow-y: auto;
  padding: 16px;
  height: inherit;
  white-space: normal;
}

<div
  class="emotion-0 emotion-1 emotion-2"
  data-testid="popoverContent"
  tabindex="-1"
  width="auto"
>
  <div
    aria-labelledby="popoverWithInverseStyles_trigger"
    role="dialog"
  >
    <div
      aria-live="polite"
      class="emotion-3 emotion-4"
      id="popoverWithInverseStyles_content"
      style="max-height: 100%;"
    >
      <div
        class="emotion-5 emotion-6"
      >
        <span
          theme="[object Object]"
        >
          Content
        </span>
      </div>
    </div>
  </div>
</div>
`;

exports[`Popover should render the popover with pointer by default 1`] = `
.emotion-1 {
  background: #FFFFFF;
  border: 1px solid #D4D4D4;
  border-radius: 8px;
  box-shadow: 0 2px 6px 0 rgba(0,0,0,0.18);
  color: #454545;
  font-family: "Work Sans",Helvetica,sans-serif;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  overflow: visible;
  padding-left: 0;
  position: relative;
  text-align: left;
  width: auto;
  position: relative;
  background: #FFFFFF;
  display: none;
  max-height: 100%;
  opacity: 0;
  outline: 0;
  -webkit-transition: opacity 0.3s;
  transition: opacity 0.3s;
  white-space: nowrap;
  border: 1px solid #D4D4D4;
  width: 100%;
  max-width: 300px;
}

.emotion-3 {
  width: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.emotion-5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  overflow-y: auto;
  padding: 16px;
  height: inherit;
  white-space: normal;
}

.emotion-1 {
  background: #FFFFFF;
  border: 1px solid #D4D4D4;
  border-radius: 8px;
  box-shadow: 0 2px 6px 0 rgba(0,0,0,0.18);
  color: #454545;
  font-family: "Work Sans",Helvetica,sans-serif;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  overflow: visible;
  padding-left: 0;
  position: relative;
  text-align: left;
  width: auto;
  position: relative;
  background: #FFFFFF;
  display: none;
  max-height: 100%;
  opacity: 0;
  outline: 0;
  -webkit-transition: opacity 0.3s;
  transition: opacity 0.3s;
  white-space: nowrap;
  border: 1px solid #D4D4D4;
  width: 100%;
  max-width: 300px;
}

.emotion-3 {
  width: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.emotion-5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  overflow-y: auto;
  padding: 16px;
  height: inherit;
  white-space: normal;
}

<div
  class="emotion-0 emotion-1 emotion-2"
  data-testid="popoverContent"
  tabindex="-1"
  width="auto"
>
  <div
    aria-labelledby="popoverWithPointer_trigger"
    role="dialog"
  >
    <div
      aria-live="polite"
      class="emotion-3 emotion-4"
      id="popoverWithPointer_content"
      style="max-height: 100%;"
    >
      <div
        class="emotion-5 emotion-6"
      >
        <span
          theme="[object Object]"
        >
          Content
        </span>
      </div>
    </div>
  </div>
</div>
`;

exports[`Popover should render the popover with target width 1`] = `
.emotion-1 {
  background: #FFFFFF;
  border: 1px solid #D4D4D4;
  border-radius: 8px;
  box-shadow: 0 2px 6px 0 rgba(0,0,0,0.18);
  color: #454545;
  font-family: "Work Sans",Helvetica,sans-serif;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  overflow: visible;
  padding-left: 0;
  position: relative;
  text-align: left;
  width: 0px;
  position: relative;
  background: #FFFFFF;
  display: none;
  max-height: 100%;
  opacity: 0;
  outline: 0;
  -webkit-transition: opacity 0.3s;
  transition: opacity 0.3s;
  white-space: nowrap;
  border: 1px solid #D4D4D4;
  width: 0px;
  max-width: 300px;
}

.emotion-3 {
  width: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.emotion-5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  overflow-y: auto;
  padding: 16px;
  height: inherit;
  white-space: normal;
}

.emotion-1 {
  background: #FFFFFF;
  border: 1px solid #D4D4D4;
  border-radius: 8px;
  box-shadow: 0 2px 6px 0 rgba(0,0,0,0.18);
  color: #454545;
  font-family: "Work Sans",Helvetica,sans-serif;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  overflow: visible;
  padding-left: 0;
  position: relative;
  text-align: left;
  width: 0px;
  position: relative;
  background: #FFFFFF;
  display: none;
  max-height: 100%;
  opacity: 0;
  outline: 0;
  -webkit-transition: opacity 0.3s;
  transition: opacity 0.3s;
  white-space: nowrap;
  border: 1px solid #D4D4D4;
  width: 0px;
  max-width: 300px;
}

.emotion-3 {
  width: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.emotion-5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  overflow-y: auto;
  padding: 16px;
  height: inherit;
  white-space: normal;
}

<div
  class="emotion-0 emotion-1 emotion-2"
  data-testid="popoverContent"
  tabindex="-1"
  width="0px"
>
  <div
    aria-labelledby="popoverWithTargetWidth_trigger"
    role="dialog"
  >
    <div
      aria-live="polite"
      class="emotion-3 emotion-4"
      id="popoverWithTargetWidth_content"
      style="max-height: 100%;"
    >
      <div
        class="emotion-5 emotion-6"
      >
        <span
          theme="[object Object]"
        >
          Content
        </span>
      </div>
    </div>
  </div>
</div>
`;

exports[`Popover should render the popover without pointer 1`] = `
.emotion-1 {
  background: #FFFFFF;
  border: 1px solid #D4D4D4;
  border-radius: 8px;
  box-shadow: 0 2px 6px 0 rgba(0,0,0,0.18);
  color: #454545;
  font-family: "Work Sans",Helvetica,sans-serif;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  overflow: visible;
  padding-left: 0;
  position: relative;
  text-align: left;
  width: auto;
  position: relative;
  background: #FFFFFF;
  display: none;
  max-height: 100%;
  opacity: 0;
  outline: 0;
  -webkit-transition: opacity 0.3s;
  transition: opacity 0.3s;
  white-space: nowrap;
  border: 1px solid #D4D4D4;
  width: 100%;
  max-width: 300px;
}

.emotion-3 {
  width: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.emotion-5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  overflow-y: auto;
  padding: 16px;
  height: inherit;
  white-space: normal;
}

.emotion-1 {
  background: #FFFFFF;
  border: 1px solid #D4D4D4;
  border-radius: 8px;
  box-shadow: 0 2px 6px 0 rgba(0,0,0,0.18);
  color: #454545;
  font-family: "Work Sans",Helvetica,sans-serif;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  overflow: visible;
  padding-left: 0;
  position: relative;
  text-align: left;
  width: auto;
  position: relative;
  background: #FFFFFF;
  display: none;
  max-height: 100%;
  opacity: 0;
  outline: 0;
  -webkit-transition: opacity 0.3s;
  transition: opacity 0.3s;
  white-space: nowrap;
  border: 1px solid #D4D4D4;
  width: 100%;
  max-width: 300px;
}

.emotion-3 {
  width: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.emotion-5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  overflow-y: auto;
  padding: 16px;
  height: inherit;
  white-space: normal;
}

<div
  class="emotion-0 emotion-1 emotion-2"
  data-testid="popoverContent"
  tabindex="-1"
  width="auto"
>
  <div
    aria-labelledby="popoverWithoutPointer_trigger"
    role="dialog"
  >
    <div
      aria-live="polite"
      class="emotion-3 emotion-4"
      id="popoverWithoutPointer_content"
      style="max-height: 100%;"
    >
      <div
        class="emotion-5 emotion-6"
      >
        <span
          theme="[object Object]"
        >
          Content
        </span>
      </div>
    </div>
  </div>
</div>
`;
